erpnext/stock/report/stock_ledger/stock_ledger.py

```diff
@@ -70,7 +71,10 @@ def update_available_serial_nos(available_serial_nos, sle):
 	serial_nos = get_serial_nos(sle.serial_no)
 	key = (sle.item_code, sle.warehouse)
 	if key not in available_serial_nos:
-		available_serial_nos.setdefault(key, [])
+		stock_balance = get_stock_balance_for(sle.item_code, sle.warehouse, sle.date.split(' ')[0],
+			sle.date.split(' ')[1], batch_no=sle.batch_no)
```

Comment:
```suggestion
			sle.date.split(' ')[1])
```

Not sure why batch no filter is required here 🤔